### PR TITLE
[INTER-166] Setup shared workflows and semantic-release

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

File: .releaserc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+              "type": "ci",
+              "section": "Continuous Integration",
+              "hidden": true
+            }
+          ]
+        }
+      }
+    ],
+    "@semantic-release/changelog",
+    "@semantic-release/npm",
```
Comment (ilfa) [inline .releaserc:103]: Do we need to publish to npm?

File: package.json [hunk 2]
```diff
@@ -32,26 +39,36 @@
...
-    "@actions/core": "^1.10.0",
-    "@actions/github": "^5.1.1",
     "@azure/arm-frontdoor": "^5.1.0",
+    "@commitlint/config-conventional": "^17.0.3",
     "@playwright/test": "^1.32.3",
     "@rollup/plugin-commonjs": "^25.0.0",
     "@rollup/plugin-json": "^6.0.0",
     "@rollup/plugin-node-resolve": "^15.0.2",
     "@rollup/plugin-replace": "^5.0.1",
+    "@semantic-release/changelog": "6.0.1",
```
Comment (ilfa) [inline package.json:49]: We don't need all of them if we use github action

**Review by ilfa — CHANGES_REQUESTED** [review]
We need to remove semantic-release deps

File: .releaserc (hunk 1 above)
Comment (TheUnderScorer) [reply .releaserc:103]: Not really, the npm plugin is also responsible for bumping version in `package.json`. It won't publish if the package is set to private.

Comment (ilfa) [reply .releaserc:103]: We can ask plugin just bump the version without publishing https://github.com/fingerprintjs/fingerprint-pro-segment-source-function/blob/main/.releaserc#L106

Comment (TheUnderScorer) [reply .releaserc]: It should work also with just setting `private` to `true` in `package.json`, but I'll set it as well 😁

**Review by ilfa — APPROVED** [review]

**Pull request merged**